src/intrinsics/ecma262/Object.js

```diff
@@ -363,9 +363,11 @@ export default function(realm: Realm): NativeFunctionValue {
     // If we're in pure scope and the items are completely abstract,
     // then create an abstract temporal with an array kind
     if (realm.isInPureScope() && obj instanceof AbstractObjectValue) {
+      // Use original object because Object.keys() call does an implicit conversion
```

Comment:
The fact that obj is an AbstractObjectValue does not tell us that the implicit conversion did not have side effects. Even if these side effects are benign and private to the current pure scope, repeating them might break things.